SignalMessaging/attachments/SignalAttachment.swift

```diff
@@ -791,8 +830,17 @@ public class SignalAttachment: NSObject {
     //       Check the attachment's error property.
     @objc
     public class func attachment(dataSource: DataSource?, dataUTI: String) -> SignalAttachment {
+        return attachment(dataSource: dataSource, dataUTI: dataUTI, imageQuality: .original)
+    }
+
+    // Factory method for attachments of any kind.
+    //
+    // NOTE: The attachment returned by this method may not be valid.
+    //       Check the attachment's error property.
+    @objc
+    public class func attachment(dataSource: DataSource?, dataUTI: String, imageQuality: TSImageQuality = .original) -> SignalAttachment {
```

Comment:
Can we default to `TSImageQuality.unspecified` (or `nil`) instead of `.original`, and change the below to something like:

```
if inputImageUTISet.contains(dataUTI) {
    guard imageQuality != .unspecified else {
      owsFail("must specify image quality type")
      imageQuality = .original
    } 
    return imageAttachment(dataSource : dataSource, dataUTI : dataUTI, imageQuality:imageQuality)
} else if videoUTISet.contains(dataUTI) {
...
```

Reply:
Obj-C interop won't let us specify optional enum parameters with default values.  I've adopted a slightly different approach of having two methods with slightly different signatures (poor man's default parameter value) and moving this kind of check there.  Should yield the behavior you desire.